src/main/java/io/apimatic/core/ErrorCase.java

```diff
@@ -95,13 +92,33 @@ public static <ExceptionType extends CoreApiException> ErrorCase<ExceptionType>
         return new ErrorCase<ExceptionType>(reason, exceptionCreator, true);
     }
 
-    private void replacePlaceHolder(Response response) {
-        replaceStatusCodeFromTemplate(response.getStatusCode());
-        replaceHeadersFromTemplate(response.getHeaders());
-        replaceBodyFromTemplate(response.getBody());
+    private String getReason(Response response) {
+        if (!isErrorTemplate) {
+            return reason;
+        }
+        String newReason = replaceStatusCodeFromTemplate(response.getStatusCode(), this.reason);
```

Comment:
Nits, Blocker: `newReason` variable name should be renamed to `resolvedReason`.

Reply:
Fixed